src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1320,6 +1331,96 @@ private bool OnlyWhitespaceOrCommentsAfterExtent(InternalScriptExtent extent)
             return true;
         }
 
+        internal bool IsPipeContinuance(IScriptExtent extent)
+        {
+            var scriptExtent = (InternalScriptExtent)extent;
+            return scriptExtent.EndOffset < _script.Length && PipeContinuanceAfterExtent(scriptExtent);
+        }
+
+        private bool PipeContinuanceAfterExtent(InternalScriptExtent extent)
+        {
+            // If the first non-newline, non-whitespace, non-comment, non-backtick, non-semi-colon character
+            // following the newline is a pipe, we have a pipe continuance.
+            bool lastNonWhitespaceIsNewline = true;
+
+            for (int i = extent.EndOffset; i < _script.Length - 1;)
```

Comment:
Agreed. It was a `for` loop because that made sense, until the logic was changed during the review process, and `while` is a better choice now, so I made that change. I also added a comment explaining the `-1`.